### PR TITLE
Verify that release notes match Release Announcement

### PR DESCRIPTION
**Pull request by quodlibetor**
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3769)

### CODE REVIEW

File: doc/user/content/release-notes.md [hunk 1]
```diff
@@ -55,13 +55,27 @@ Wrap your release notes at the 80 character mark.
...
+- Many stability improvements, including a chaos-testing framework
+- Support for Confluent Schema Registry
+- Change Data Capture format updates which allow Kafka sinks to optionally emit Debezium
+  style, with new documentation about our [consistency
+  metadata](/sql/create-sink/#consistency-metadata).
+- Time Travel via [AS OF](/docs/sql/create-sink/#as-of).
+- Introduce the ability to [rename indexes, sinks, sources, and
+  views](/sql/alter-rename).
+- Released [our Avro library as open source][mz-avro] (Apache 2.0).
+- Released a way to tail database binlogs and emit CDC data without Kafka, [tb][]
```
[the remark this answers is not in the document]
Comment (awang) [reply doc/user/content/release-notes.md:70]: They're both in the blog post!  https://materialize.io/release-materialize-0-4/

File: doc/user/content/release-notes.md [hunk 2]
```diff
@@ -55,13 +55,27 @@ Wrap your release notes at the 80 character mark.
 <span id="v0.4.0"></span>
 ## v0.4.0
 
-- Introduce the ability to [rename indexes, sinks, sources, and views](/sql/alter-rename).
+Read the [Release Announcement](https://materialize.io/release-materialize-0-4/) for more
+details.
+
+- Many stability improvements, including a chaos-testing framework
+- Support for Confluent Schema Registry
```
[the remark this answers is not in the document]
Comment (awang) [reply doc/user/content/release-notes.md:62]: We added SASL Plain support, which made it possible to use Materialize with Confluent Cloud: https://github.com/MaterializeInc/materialize/commit/63c3fcdd2b8bbc356a5362a842bb6774919f0e0f

(this stems from a mistake on my part; I've updated the blog post)

**Comment by quodlibetor** [comment]
> My slight apprehension here is that the release notes might have a more atomized list of "features" that are coalesced into one über feature in the blog post.

Yeah I agree with this. Right now our issue is that relnotes sometimes have fewer features than the release announcement, which seems wrong. I'll try rewording.

File: doc/user/content/release-notes.md [hunk 3]
```diff
@@ -55,13 +55,27 @@ Wrap your release notes at the 80 character mark.
...
+- Introduce the ability to [rename indexes, sinks, sources, and
+  views](/sql/alter-rename).
+- Released [our Avro library as open source][mz-avro] (Apache 2.0).
+- Released a way to tail database binlogs and emit CDC data without Kafka, [tb][]
 - Rename the `-w`/`--threads` command line argument to `-w`/`--workers`, since it
   reflects timely workers and does not limit the number of threads that materialized may
   start.
 - Fix a bug that prevented dropping databases with cross-schema dependencies.
   {{% gh 3558 %}}
-- Allow Kafka sinks to optionally emit Debezium style [consistency metadata](/sql/create-sink/#consistency-metadata).
```
[the remark this answers is not in the document]
Comment (quodlibetor) [reply doc/user/content/release-notes.md:64]: To make it more obviously associated with the same item in the release announcement.

File: doc/user/content/release-notes.md [hunk 4]
```diff
@@ -55,13 +55,27 @@ Wrap your release notes at the 80 character mark.
 <span id="v0.4.0"></span>
 ## v0.4.0
 
-- Introduce the ability to [rename indexes, sinks, sources, and views](/sql/alter-rename).
+Read the [Release Announcement](https://materialize.io/release-materialize-0-4/) for more
```
[the remark this answers is not in the document]
Comment (quodlibetor) [reply doc/user/content/release-notes.md:58]: Yeah I waffled back and forth on this. I'll remove it.

File: doc/user/content/release-notes.md [hunk 5]
```diff
@@ -55,13 +55,27 @@ Wrap your release notes at the 80 character mark.
...
+details.
+
+- Many stability improvements, including a chaos-testing framework
```
[the remark this answers is not in the document]
Comment (quodlibetor) [reply doc/user/content/release-notes.md:61]: I can drop this.

File: doc/user/content/release-notes.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (quodlibetor) [reply doc/user/content/release-notes.md:70]: mz-avro I agree is not a change to Materialize itself, but it is a dramatic increase in the API surface area of things that we maintain.

tb is a major change in the way that folks can integrate with us, and so in the way that it's possible for folks to operate materialize, which is an external API, and is another thing that we are committing to maintain the API of.

I think both are worth calling out here, although we could easily say that future updates for their releases will be in their repos/directories.

File: doc/user/content/release-notes.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (awang) [reply doc/user/content/release-notes.md:61]: I agree! writing this 0.4 release blog post was tough for me in that there were lots of improvements that were too granular for a blog post. They might be interesting as release notes though?  I couldn't grok everything we improved, but happy to take another pass if we discuss

File: doc/user/content/release-notes.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (quodlibetor) [reply doc/user/content/release-notes.md:70]: I deleted them, though.

File: doc/user/content/release-notes.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (quodlibetor) [reply doc/user/content/release-notes.md]: Yeah I think in general we should have a stronger culture of "did you add this PR to the release notes" as part of review.

**Pull request merged**